### PR TITLE
Add `unknown-lints` lint

### PR DESCRIPTION
**Pull request by Muscraft**
This PR adds a lint for `unknown-lints` similar [`rustc::unknown-lints`](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#unknown-lints). This will help users know when they are passing something to `cargo` that `cargo` doesn't understand.

### CODE REVIEW

File: src/cargo/util/lints.rs [hunk 1]
```diff
@@ -136,6 +138,121 @@ impl From<TomlLintLevel> for LintLevel {
...
+        .collect::<HashSet<_>>();
+
+    if let Some(lints) = original_toml
+        .workspace
+        .as_ref()
+        .map_or(&None, |ws| &ws.lints)
+    {
+        if let Some(cargo_lints) = lints.get("cargo") {
+            for (name, _) in cargo_lints.iter() {
+                let normalized_name = name.replace("-", "_");
```
Comment (epage) [inline src/cargo/util/lints.rs:188]: This is having me lean more towards normalizing earlier in the process

File: src/cargo/util/lints.rs [hunk 2]
```diff
@@ -136,6 +138,121 @@ impl From<TomlLintLevel> for LintLevel {
...
+    let all_names = LINTS
+        .iter()
+        .map(|lint| lint.name)
+        .chain(LINT_GROUPS.iter().map(|group| group.name))
+        .collect::<HashSet<_>>();
+
+    if let Some(lints) = original_toml
+        .workspace
+        .as_ref()
+        .map_or(&None, |ws| &ws.lints)
```
Comment (epage) [inline src/cargo/util/lints.rs:184]: imo this should be treated as a separate "workspace" lint that is only subject to `workspace.lints` so we have consistency between virtual and real workspace packages.

File: tests/testsuite/lints_table.rs [hunk 3]
```diff
@@ -763,7 +763,6 @@ fn cargo_lints_nightly_required() {
                 authors = []
 
                 [lints.cargo]
-                "unused-features" = "deny"
```
Comment (epage) [inline tests/testsuite/lints_table.rs:766]: Why were these here and why are they being removed?

**Comment by ehuss** [comment]
r? epage
if you don't mind.

**Comment by Muscraft** [comment]
Going to close in favor of #13837, as it contains the addition of an `unknown_lints`
lint
